test/integration/base.py

```diff
@@ -26,30 +23,11 @@ def set_http_debug(self):
             http.client.HTTPConnection.debuglevel = 1
 
     @pytest.fixture(autouse=True)
-    def save_settings(self, dont_cleanup_old_buckets, b2_auth_data):
-        type(self).dont_cleanup_old_buckets = dont_cleanup_old_buckets
-        type(self).b2_auth_data = b2_auth_data
-
-    @classmethod
-    def setup_class(cls):
-        cls.this_run_bucket_name_prefix = GENERAL_BUCKET_NAME_PREFIX + bucket_name_part(8)
```

Comment:
Old bucket creation also ensured that there is additional, random part of the prefix for each test. This way, during cleanup, we could remove buckets that belongs to this test session only. Other sessions, running in parallel, would be unaffected. Is there something like this in the new code and I just missed it?

Reply:
Here, instead of this flag, we use a mixin that holds a list of buckets created within a given class instance. The clean_buckets function goes through the list of buckets created only for that instance and removes them. This way we don't have to keep a prefix for each session